app/src/main/kotlin/com/softteco/template/ui/feature/bluetooth/BluetoothScreen.kt

```diff
@@ -0,0 +1,139 @@
+import com.softteco.template.ui.theme.Dimens.PaddingExtraSmall
+import com.softteco.template.ui.theme.Dimens.PaddingSmall
+
+@Composable
+fun BluetoothScreen(
+    modifier: Modifier = Modifier,
+    viewModel: BluetoothViewModel = hiltViewModel(),
+    onBackClicked: () -> Unit = {}
+) {
+    val state by viewModel.state.collectAsState()
+
+    ScreenContent(
+        state,
+        modifier,
+        onBackClicked
+    )
+}
+
+@Composable
+private fun ScreenContent(
+    state: BluetoothViewModel.State,
+    modifier: Modifier = Modifier,
+    onBackClicked: () -> Unit = {}
+) {
+    TextSnackbarContainer(
+        modifier = modifier,
+        snackbarText = stringResource(state.snackBar.textId),
+        showSnackbar = state.snackBar.show,
+        onDismissSnackbar = state.dismissSnackBar
+    ) {
+        Column(modifier = Modifier.fillMaxWidth()) {
+            CustomTopAppBar(
+                stringResource(id = R.string.bluetooth),
+                showBackIcon = true,
+                modifier = Modifier.fillMaxWidth(),
+                onBackClicked = onBackClicked
+            )
+            BluetoothDevicesList(state)
+        }
+    }
+}
+
+@Composable
+private fun BluetoothDevicesList(
+    state: BluetoothViewModel.State
+) {
+    LazyColumn(
+        modifier = Modifier.fillMaxWidth(),
+        contentPadding = PaddingValues(PaddingDefault)
+    ) {
+        items(state.bluetoothDevices) { bluetoothDevice ->
+            BluetoothDeviceCard(
+                bluetoothDevice.name,
+                bluetoothDevice.macAddress,
+                bluetoothDevice.rssi
+            )
+        }
+    }
+}
+
+@Composable
+private fun BluetoothDeviceCard(
+    name: String,
+    macAddress: String,
+    rssi: Int
+) {
+    Card(
+        modifier = Modifier
+            .padding(PaddingDefault)
+            .fillMaxWidth()
+            .wrapContentHeight(),
+        shape = MaterialTheme.shapes.medium
+    ) {
+        Row(verticalAlignment = Alignment.CenterVertically) {
+            Image(
+                painter = painterResource(R.drawable.ic_bluetooth),
+                contentDescription = "ic_bluetooth",
+                modifier = Modifier
+                    .padding(PaddingSmall),
+                contentScale = ContentScale.Fit
+            )
+            Column(Modifier.padding(PaddingSmall)) {
+                Text(text = name)
+                Text(text = macAddress)
+                Row(verticalAlignment = Alignment.CenterVertically) {
+                    Image(
+                        painter = painterResource(R.drawable.ic_signal_level),
+                        contentDescription = "ic_bluetooth",
+                        modifier = Modifier
+                            .padding(PaddingExtraSmall),
+                        contentScale = ContentScale.Fit
+                    )
+                    Text(text = rssi.toString())
+                }
+            }
+            Column(
```

Comment:
It is possible to just use a `Spacer` with a `weight`, instead of wrapping the button in a `Column`